### PR TITLE
Uniqueness and existential uniqueness

### PR DESCRIPTION
**Pull request by benjub**
Continue reordering. See two commit messages.

I moved curryax to Main since I used it at some point, but its use was then removed.  Anyway, I think it is of independent interest. Drawing @jkingdon's attention to curryax and the comments of peirce/curryax.

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -22561,16 +22578,51 @@ of the unique existential quantifier (note that ` y ` and ` z ` need not
...
   $}
 
-  $( Alternate definition of the at-most-one quantifier, in terms of the
+  $( Existential uniqueness implies existence.  (Contributed by NM,
+     15-Sep-1993.)  (Proof shortened by Andrew Salmon, 9-Jul-2011.)  (Proof
+     shortened by Wolf Lammen, 4-Dec-2018.)  (Proof shortened by BJ,
+     7-Oct-2022.) $)
+  euex $p |- ( E! x ph -> E. x ph ) $=
+    ( weu wex wmo df-eu simplbi ) ABCABDABEABFG $.
+  $( Existential uniqueness implies uniqueness.  (Contributed by NM,
```
Comment (avekens) [inline set.mm:22587]: empty line missing

File: set.mm [hunk 2]
```diff
@@ -22632,89 +22703,142 @@ of the unique existential quantifier (note that ` y ` and ` z ` need not
...
+       of ` x ` .  This is ensured by having ` x ` and ` y ` be distinct.
+       Alternately, a distinctor ` -. A. x x = y ` could have been used
+       instead.  See ~ eueq and ~ eueqi for classes.  (Contributed by Stefan
+       Allan, 4-Dec-2008.)  (Proof shortened by Wolf Lammen, 8-Sep-2019.) $)
+    euequ $p |- E! x x = y $=
+      ( vz weq weu wb wal wex ax6evr equequ2 alrimiv eximii eu6 mpbir ) ABDZAEO
+      ACDFZAGZCHBCDZQCCBIRPABCAJKLOACMN $.
+  $}
+
+  $( old theorems $)
```
Comment (avekens) [inline set.mm:22769]: I think this comment can be removed.

File: set.mm (hunk 1 above)
Comment (avekens) [reply set.mm:22587]: This seems to be fixed meanwhile...

**Review by avekens — APPROVED** [review]

File: set.mm (hunk 2 above)
Comment (benjub) [reply set.mm:22769]: I planned on removing it (and the comment `($ substitution $)`) when the whole section is treated. Maybe two or three more PRs.  In the meantime, it helps me keep track of what has been done. Is that ok with you ?

File: set.mm [hunk 3]
```diff
@@ -22426,6 +22430,63 @@ primitive connectives (no defined terms) on the right-hand side.
       FBTUDCAUCJKLUDBCMNOABCPRS $.
   $}
 
+  $( Absorption of existence condition by uniqueness.  (Contributed by NM,
+     4-Nov-2002.)  Shorten proof and avoid ~ df-eu .  (Revised by BJ,
+     14-Oct-2022.) $)
+  moabs $p |- ( E* x ph <-> ( E. x ph -> E* x ph ) ) $=
+    ( wmo wex wi ax-1 wn nexmo pm2.6 ax-mp impbii ) ABCZABDZLEZLMFMGLENLEABHMLI
+    JK $.
```
Comment (wlammen) [inline set.mm:22438]: Suggest to prove this using nexmo, id, ja instead of nexmo, pm2.6, ax-mp

File: set.mm [hunk 4]
```diff
@@ -22426,6 +22430,63 @@ primitive connectives (no defined terms) on the right-hand side.
...
+
+  $( Any proposition holds for some ` x ` or holds for at most one ` x ` .
+     (Contributed by NM, 8-Mar-1995.)  Shorten proof and avoid ~ df-eu .
+     (Revised by BJ, 14-Oct-2022.) $)
+  exmo $p |- ( E. x ph \/ E* x ph ) $=
+    ( wex wmo wo wn wi nexmo df-or mpbir ) ABCZABDZEKFLGABHKLIJ $.
```
Comment (wlammen) [inline set.mm:22444]: suggest to prove this using orri

File: set.mm [hunk 5]
```diff
@@ -22426,6 +22430,63 @@ primitive connectives (no defined terms) on the right-hand side.
...
+    $d x z $.  $d y z $.  $d z ph $.  $d z ps $.
+    nfmod2.1 $e |- F/ y ph $.
+    nfmod2.2 $e |- ( ( ph /\ -. A. x x = y ) -> F/ x ps ) $.
+    $( Bound-variable hypothesis builder for the at-most-one quantifier.
+       (Contributed by Mario Carneiro, 14-Nov-2016.)  Avoid ~ df-eu .  (Revised
+       by BJ, 14-Oct-2022.) $)
+    nfmod2 $p |- ( ph -> F/ x E* y ps ) $=
+      ( vz wmo weq wi wal wex df-mo nfv wn wa wnf nfeqf1 adantl nfimd nfald2
+      nfexd nfxfrd ) BDHBDGIZJZDKZGLACBDGMAUFCGAGNAUECDEACDICKOZPBUDCFUGUDCQACD
+      GRSTUAUBUC $.
```
Comment (wlammen) [inline set.mm:22456]: I don't see an advantage of this proof over the old one.  See orim12d/ALT for a similar situation.  This is not a consistent view.

**Review by wlammen — CHANGES_REQUESTED** [review]

File: set.mm [hunk 6]
```diff
@@ -22426,6 +22430,63 @@ primitive connectives (no defined terms) on the right-hand side.
...
+  ${
+    $d x y $.  $d y ph $.  $d y ps $.  $d y ch $.
+    mobid.1 $e |- F/ x ph $.
+    mobid.2 $e |- ( ph -> ( ps <-> ch ) ) $.
+    $( Formula-building rule for the at-most-one quantifier (deduction form).
+       (Contributed by NM, 8-Mar-1995.)  Avoid ~ df-eu .  (Revised by BJ,
+       14-Oct-2022.) $)
+    mobid $p |- ( ph -> ( E* x ps <-> E* x ch ) ) $=
+      ( vy weq wi wal wex wmo imbi1d albid exbidv df-mo 3bitr4g ) ABDGHZIZDJZGK
+      CRIZDJZGKBDLCDLATUBGASUADEABCRFMNOBDGPCDGPQ $.
```
Comment (wlammen) [inline set.mm:22487]: This is a reduction in axiom usage as well (ax-10, ax-11, ax-13 not needed)

File: set.mm (hunk 4 above)
Comment (benjub) [reply set.mm:22444]: Thanks ! I apparently forgot to run `MM-PA> minimize_with *`

File: set.mm (hunk 3 above)
Comment (benjub) [reply set.mm:22438]: Thanks ! `MM-PA> minimize_with *` did not catch that. Do you want I add the "proof shortened by WL" tag ?

Comment (wlammen) [reply set.mm:22438]: no

**Review by wlammen — APPROVED** [review]

**Pull request merged**